larpandoracontent/LArTrackShowerId/SvmPfoCharacterisationAlgorithm.h

```diff
@@ -26,21 +26,28 @@ class SvmPfoCharacterisationAlgorithm : public PfoCharacterisationBaseAlgorithm
     SvmPfoCharacterisationAlgorithm();
 
 protected:
+    virtual bool IsClearTrack(const pandora::ParticleFlowObject *const pPfo) const;
     virtual bool IsClearTrack(const pandora::Cluster *const pCluster) const;
     pandora::StatusCode ReadSettings(const pandora::TiXmlHandle xmlHandle);
 
-    ClusterCharacterisationFeatureTool::FeatureToolVector   m_featureToolVector;    ///< The feature tool map
-    SupportVectorMachine    m_supportVectorMachine;         ///< The support vector machine
+    ClusterCharacterisationFeatureTool::FeatureToolVector   m_featureToolVector;         ///< The feature tool map
+    PfoCharacterisationFeatureTool::FeatureToolVector       m_featureToolVectorThreeD;   ///< The feature tool map for 3D info
+    PfoCharacterisationFeatureTool::FeatureToolVector       m_featureToolVectorMissingW; ///< The feature tool map for missing W view
+   
```

Comment:
Going back to earlier comments, we should ask whether we still want to run the 3x2D version here, or whether we could simplify everything (including the base class interface!) by just now running with Pfos and in 3D. Are there any planned usages of the 3x2D approach still?

Removing all the 3x2D stuff, including the IsClearTrack(const Cluster*) callback would definitely tidy things up. An intermediate case might be to add two different algorithms, rather than one algorithm that has a single ShouldUseThreeDInfo setting.

Reply:
under discussion